File: backend/backend/src/main/java/com/flightpriceanalytics/backend/dto/InitResponse.java
package com.flightpriceanalytics.backend.dto;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.List;

public record InitResponse(
        Route route,
        Period period,
        String currency,
        Instant lastUpdatedAt,
        Today today,
        List<DailyPrice> dailyPrices,
        List<AirlineRanking> airlineRanking,
        List<DepartureTimeBand> departureTimeBands
) {
    public record Route(String origin, String destination) {
    }

    public record Period(LocalDate from, LocalDate to, String timezone) {
    }

    public record Today(
            LocalDate departureDate,
            long offerCount,
            BigDecimal averagePrice,
            BigDecimal lowestPrice,
            BigDecimal highestPrice
    ) {
    }

    public record DailyPrice(LocalDate date, BigDecimal averagePrice, long offerCount) {
    }

    public record AirlineRanking(
            String carrierCode,
            String carrierName,
            BigDecimal averagePrice,
            long offerCount
    ) {
    }

    public record DepartureTimeBand(String timeBand, BigDecimal averagePrice, long offerCount) {
    }
}

File: frontend/src/types/initResponse.ts
/** YYYY-MM-DD形式の日付文字列 */
export type IsoDate = string;

/** ISO 8601形式の日時文字列 */
export type IsoDateTime = string;

export type DepartureTimeBandName =
  | "earlyMorning"
  | "morning"
  | "afternoon"
  | "evening";

export interface Route {
  origin: string;
  destination: string;
}

export interface Period {
  from: IsoDate;
  to: IsoDate;
  timezone: string;
}

export interface Today {
  departureDate: IsoDate;
  offerCount: number;
  averagePrice: number | null;
  lowestPrice: number | null;
  highestPrice: number | null;
}

export interface DailyPrice {
  date: IsoDate;
  averagePrice: number | null;
  offerCount: number;
}

export interface AirlineRanking {
  carrierCode: string;
  carrierName: string;
  averagePrice: number;
  offerCount: number;
}

export interface DepartureTimeBand {
  timeBand: DepartureTimeBandName;
  averagePrice: number | null;
  offerCount: number;
}

export interface InitResponse {
  route: Route;
  period: Period;
  currency: string;
  lastUpdatedAt: IsoDateTime;
  today: Today;
  dailyPrices: DailyPrice[];
  airlineRanking: AirlineRanking[];
  departureTimeBands: DepartureTimeBand[];
}
